flask_user/views.py

```diff
@@ -440,13 +440,11 @@ def register():
                                      user=user,
                                      user_invite=user_invite)
 
-        # Redirect if USER_ENABLE_CONFIRM_EMAIL is set
-        if user_manager.enable_confirm_email and require_email_confirmation:
-            next = request.args.get('next', _endpoint_url(user_manager.after_register_endpoint))
-            return redirect(next)
+        # Redirect if USER_ENABLE_CONFIRM_EMAIL is set and they can't login yet
+        if user_manager.enable_confirm_email and require_email_confirmation and not user_manager.enable_login_without_confirm_email:
+            return redirect(reg_next)
```

Comment:
I'm not sure if this things have to be in the same pull request with `USER_URL_PREFIX`. I would prefer to have a separate PR with this changes. Any way thank you! It looks useful too! :+1: